src/main/java/com/puppycrawl/tools/checkstyle/checks/design/DesignForExtensionCheck.java

```diff
@@ -208,6 +216,58 @@
  * }
  * </pre>
  * <p>
+ * To configure the check to allow methods which contain a specified comment text
+ * pattern in their javadoc to be designed for extension.
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;DesignForExtension&quot;&gt;
+ *   &lt;property name=&quot;requiredJavadocPhrase&quot;
+ *     value=&quot;This implementation&quot;/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <pre>
+ * public class A {
+ *   /&#42;&#42;
+ *   &#42; This implementation ...
+ *   &#42;/
+ *   public int foo() {return 2;} // ok, required javadoc phrase in comment
+ *
+ *   /&#42;&#42;
+ *   &#42; Do not extend ...
+ *   &#42;/
+ *   public int foo2() {return 8;} // violation, required javadoc phrase not in comment
+ *
+ *   public int foo3() {return 3;} // violation, required javadoc phrase not in comment
+ * }
+ * </pre>
+ * <p>
+ * To configure the check to allow methods which contain a specified comment text
+ * pattern in their javadoc which can span multiple lines
```

Comment:
What happens if there is HTML between the text? Bold for example. How do I configure this?

Reply:
I think it will be just text